### PR TITLE
 Use ohsome API endpoint `/contributions/latest/count` for the Last Edit indicator

### LINKED ISSUE #35: Use ohsome API endpoint `/contributions/latest/count` once it is available
This endpoint would supplant current endpoint of the Last Edit indicator and speed up the computation the indicator substantially.

See discussion of the relevant ohsome API issue: https://github.com/GIScience/ohsome-api/issues/174 

## Blocker
- [x]  https://github.com/GIScience/ohsome-api/issues/174

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Corresponding issue
Closes #35 

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

File: workers/tests/integrationtests/test_indicator_last_edit.py [hunk 1]
```diff
@@ -30,18 +30,6 @@ def test(self):
         indicator.create_figure()
         self.assertIsNotNone(indicator.result.svg)
 
-    # TODO: Choose smaller test region for this test case
```
Comment (matthiasschaub) [inline workers/tests/integrationtests/test_indicator_last_edit.py:33]: Can not reproduce this test case with this regions.

Comment (matthiasschaub) [reply workers/tests/integrationtests/test_indicator_last_edit.py:33]: Meaning that for this region element count is bigger that contributions latest count

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py [hunk 2]
```diff
@@ -23,83 +25,79 @@ def __init__(
...
-        if self.total_features == 0:
+        # It can be that features are counted, but have been deleted since.
+        if self.element_count == 0:
             self.result.label = "undefined"
             self.result.value = None
-            self.result.description = (
-                "Since the OHSOME query returned a count of 0 for this feature "
-                "a quality estimation can not be made for this filter"
-            )
+            self.result.description = "In the AOI no features are present."
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:68]: Is _AOI_ commonly used in OQT or known to the people using OQT?

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py [hunk 3]
```diff
@@ -23,83 +25,79 @@ def __init__(
...
+
+        if self.contributions_latest_count > self.element_count:
+            self.ratio = 100
+        else:
+            self.ratio = round(
+                (self.contributions_latest_count / self.element_count) * 100
+            )
+
+        description = Template(self.metadata.result_description).substitute(
+            share=self.contributions_latest_count, layer_name=self.layer.name
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:79]: Shouldn't the share be `self.ratio`?

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py [hunk 4]
```diff
@@ -23,83 +25,79 @@ def __init__(
...
             )
-        elif self.share_edited_features >= self.threshold_red:
+        elif self.ratio >= self.threshold_red:
             self.result.label = "yellow"
             self.result.value = 0.5
             self.result.description = (
                 description + self.metadata.label_description["yellow"]
             )
-        else:
+        elif self.ratio < self.threshold_red:
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:94]: Not sure what the best programming behaviour is here? If any `elif` case is ever changed here, you wouldn't have covered cases. Maybe you can add a `else:` case with an exception that shouldn't ever occur?

File: workers/tests/integrationtests/test_indicator_last_edit.py (hunk 1 above)
Comment (joker234) [reply workers/tests/integrationtests/test_indicator_last_edit.py:33]: Delete this for now.

**Review by joker234 — COMMENTED** [review]
* General Question: why do we calculate shares in `[0,100]` and not `[0,1]` ranges?
* `get_latest_ohsome_timestamp()` can have type hints as well

File: workers/tests/integrationtests/test_indicator_last_edit.py (hunk 1 above)
Comment (matthiasschaub) [reply workers/tests/integrationtests/test_indicator_last_edit.py:33]: This test case has now an issue 
https://github.com/GIScience/ohsome-quality-analyst/issues/74

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py [hunk 5]
```diff
@@ -23,83 +25,79 @@ def __init__(
...
             bpolys=bpolys,
         )
-        self.time_range = time_range
-        # TODO: thresholds might be better defined for each OSM layer
         self.threshold_yellow = 20  # more than 20% edited last year --> green
-        self.threshold_red = 5  # more than 5% edited last year --> yellow
-        self.edited_features = None
-        self.total_features = None
-        self.share_edited_features = None
+        self.threshold_red = 5
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:29]: Yes I thought one commen is enough to make clear what this threshhold means. This understanding can be transfered to all threshhold variables.

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py (hunk 2 above)
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:68]: Changed it to area of interest

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py [hunk 6]
```diff
@@ -23,83 +25,79 @@ def __init__(
...
             latest_ohsome_stamp = await ohsome_client.get_latest_ohsome_timestamp()
-            self.time_range = "{},{}".format(
-                (latest_ohsome_stamp - relativedelta(years=1)).strftime("%Y-%m-%d"),
-                latest_ohsome_stamp.strftime("%Y-%m-%d"),
+            self.time_range = "{start},{end}".format(
+                start=(latest_ohsome_stamp - relativedelta(years=1)).strftime(
+                    "%Y-%m-%d"
+                ),
+                end=latest_ohsome_stamp.strftime("%Y-%m-%d"),
             )
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:44]: Yes, thats ugly :D. I used your suggestion in the code.

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:29]: I changed the comment to be more general: 
"Threshold values are in percentage"

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py (hunk 3 above)
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:79]: I changed in here and in metadata.yml to ratio,

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py (hunk 4 above)
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py:94]: The ValueError exception will now be raised once the else branch is reached.

File: workers/ohsome_quality_analyst/indicators/last_edit/indicator.py (hunk 3 above)
Comment (joker234) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py]: I meant the value to the parameter `share` should be `self.ratio`, not the name itself ;)

Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/indicators/last_edit/indicator.py]: Oh. Thats right of course. Than this has been wrong before the changes of this PR.
I fixed this.

**Review by joker234 — APPROVED** [review]

**Pull request merged**